DQMOffline/L1Trigger/src/L1TStage2CaloLayer2Offline.cc

```diff
@@ -656,5 +692,31 @@ void L1TStage2CaloLayer2Offline::bookJetHistos(DQMStore::IBooker & ibooker)
   ibooker.cd();
 }
 
+bool L1TStage2CaloLayer2Offline::passesTrigger() const {
+  std::vector<bool> results = dqmoffline::l1t::getTriggerResults(triggerIndices_, triggerResults_);
+  if (std::count(results.begin(), results.end(), true) == 0) {
+    return false;
+  }
+  return true;
+}
+bool L1TStage2CaloLayer2Offline::passesTrigger(const l1t::Jet & jet) const{
+  // get HLT objects of fired triggers
+  using namespace dqmoffline::l1t;
+  std::vector<bool> results = getTriggerResults(triggerIndices_, triggerResults_);
+  std::vector<unsigned int> firedTriggers = getFiredTriggerIndices(triggerIndices_, results);
+  std::vector<edm::InputTag> hltFilters = getHLTFilters(firedTriggers, hltConfig_, triggerProcess_);
+  const trigger::TriggerObjectCollection hltObjects = getTriggerObjects(hltFilters, triggerEvent_);
+  // only take objects with et() > 27 GeV
+  trigger::TriggerObjectCollection filteredHltObjects;
+  std::copy_if(hltObjects.begin(), hltObjects.end(),
+             std::back_inserter(filteredHltObjects),
+             [](auto obj) { return obj.et() > 27; });
+  double l1Eta = jet.eta();
+  double l1Phi = jet.phi();
+  const trigger::TriggerObjectCollection matchedObjects = getMatchedTriggerObjects(l1Eta, l1Phi, 0.3, hltObjects);
+
+  return matchedObjects.size() == 0;
```

Comment:
Is this correct? I would think that if the jet passes the trigger the matchedObjects size should be != 0 and this function then returns false.

Reply:
If we were using a jet trigger, yes. Since we are using Muon triggers, the L1 jet and the HLT objects need to not overlap, right?
See https://its.cern.ch/jira/browse/CMSLITDPG-383?focusedCommentId=1863451&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-1863451

Reply:
Ah I see, yes. I guess I was confused by the name of the function.

Reply:
If you were confused, others will be as well. Let's change it now.
How about: `bool L1TStage2CaloLayer2Offline::doesNotOverlapWithHLTObjects(const l1t::Jet & jet) const` instead?

Reply:
Sounds good to me.